### PR TITLE
Implement DatetimeIndex.month_name() and DatetimeIndex.day_name()

### PR DESCRIPTION
**Pull request by xinrong-databricks**
```py
>>> idx = ks.date_range(start='2018-01', freq='M', periods=3)
>>> idx.month_name()
Index(['January', 'February', 'March'], dtype='object')

>>> idx = ks.date_range(start='2018-01-01', freq='D', periods=3)
>>> idx.day_name()
Index(['Monday', 'Tuesday', 'Wednesday'], dtype='object')
```

### CODE REVIEW

File: databricks/koalas/indexes/datetimes.py [hunk 1]
```diff
@@ -523,6 +523,52 @@ def round(self, freq, *args, **kwargs) -> "DatetimeIndex":
 
         return DatetimeIndex(self.to_series().dt.round(freq, *args, **kwargs))
 
+    def month_name(self, locale: Optional[str] = None) -> Index:
+        """
+        Return the month names of the DateTimeIndex with specified locale.
```
Comment (ueshin) [inline databricks/koalas/indexes/datetimes.py:528]: nit: `DatetimeIndex`?

**Review by ueshin — APPROVED** [review]
LGTM.

File: databricks/koalas/indexes/datetimes.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/indexes/datetimes.py]: Good catch! Updated.

**Review by itholic — APPROVED** [review]
Otherwise, LGTM!

**Comment by itholic** [comment]
Let me just merge this to include this change to 1.7.0 release.

Let me fix https://github.com/databricks/koalas/pull/2086#discussion_r589129019.

**Pull request merged**